packages/fx-core/src/core/tools.ts

```diff
@@ -111,26 +117,53 @@ export function isValidProject(workspacePath?: string): boolean {
   }
 }
 
-export function isV1Project(workspacePath?: string): boolean {
-  if (!workspacePath) return false;
+export async function validateV1Project(
+  workspacePath: string | undefined
+): Promise<string | undefined> {
+  if (!workspacePath) {
+    return "The workspace path cannot be empty.";
+  }
+
+  const v2ConfigFolder = path.resolve(workspacePath, `.${ConfigFolderName}`);
+  if (await fs.pathExists(v2ConfigFolder)) {
+    return `Folder '.${ConfigFolderName}' already exists.`;
+  }
+
+  const packageJsonPath = path.resolve(workspacePath, "package.json");
+  let packageSettings: any | undefined;
+
   try {
-    const confFolderPath = path.resolve(workspacePath, `.${ConfigFolderName}`);
-    if (fs.existsSync(confFolderPath)) {
-      return false;
+    packageSettings = await fs.readJson(packageJsonPath);
+  } catch (error: any) {
+    return `Cannot read 'package.json'. ${error?.message}`;
+  }
+
+  if (!packageSettings?.msteams) {
+    return "Teams Toolkit V1 settings cannot be found in 'package.json'.";
+  }
+
+  const manifestPath = path.resolve(workspacePath, AppPackageFolderName, V1ManifestFileName);
+  if (!(await fs.pathExists(manifestPath))) {
+    return "The project should be created after version 1.2.0";
```

Comment:
If any user error on migration, is there any documentation to explain the detailed prerequisites or limitations?

Reply:
Good idea, I will confirm with PM about the documents on the docs site.